### PR TITLE
[Web] Add license flag to disable inaccessible features from being discoverable in the UI

### PR DESCRIPTION
**Pull request by rudream**
## Purpose

This PR closes https://github.com/gravitational/teleport.e/issues/1803

`e` counterpart: https://github.com/gravitational/teleport.e/pull/1945

This adds a license feature flag for `featureHiding`, which hides the entire `Management` section in the UI if the user has access to none of the features within it. It also hides access requests if the user doesn't have permission to access them.

#### User who doesn't have access to any management features or access requests:
image

### CODE REVIEW

File: web/packages/teleport/src/config.ts [hunk 1]
```diff
@@ -44,6 +44,7 @@ const cfg = {
   recoveryCodesEnabled: false,
   // IsUsageBasedBilling determines if the user subscription is usage-based (pay-as-you-go).
   isUsageBasedBilling: false,
+  featureHidingEnabled: false,
```
Comment (ravicious) [inline web/packages/teleport/src/config.ts:47]: What do you think about `hideInaccessibleUi`/`hideInaccessibleFeatures`?

File: api/client/webclient/webconfig.go [hunk 2]
```diff
@@ -67,6 +67,8 @@ type WebConfig struct {
 	AutomaticUpgrades bool `json:"automaticUpgrades"`
 	// AssistEnabled is true when Teleport Assist is enabled.
 	AssistEnabled bool `json:"assistEnabled"`
+	// FeatureHidingEnabled is true when features should be undiscoverable to users without the necessary permissions.
```
Comment (ravicious) [inline api/client/webclient/webconfig.go:70]: I'd explain that usually we show certain UI elements even if the user doesn't have access to them, but with this flag being set to true, the elements are hidden instead. "Features should be undiscoverable" doesn't explain how we make them undiscoverable.

**Review by zmb3 — APPROVED** [review]

File: web/packages/teleport/src/teleportContext.tsx [hunk 3]
```diff
@@ -122,9 +122,44 @@ class TeleportContext implements types.Context {
         locks: false,
         newLocks: false,
         assist: false,
+        managementSection: false,
       };
     }
 
+    // If feature hiding is enabled in the license, this returns true if the user has no access to any features within the management section.
```
Comment (ibeckermayer) [inline web/packages/teleport/src/teleportContext.tsx:129]:
```suggestion
    // If feature hiding is enabled in the license, this returns true if the user has list access to any feature within the management section.
```

File: web/packages/teleport/src/teleportContext.tsx [hunk 4]
```diff
@@ -122,9 +122,44 @@ class TeleportContext implements types.Context {
...
+      // having list access, requestable roles, or allowed search_as_roles.
+      if (cfg.hideInaccessibleFeatures) {
+        return !!(
+          userContext.getAccessRequestAccess().list ||
+          userContext.getAccessCapabilities().requestableRoles.length ||
+          userContext.getAllowedSearchAsRoles().length
+        );
+      }
+
+      return !cfg.isDashboard;
```
Comment (ibeckermayer) [inline web/packages/teleport/src/teleportContext.tsx:160]: Maybe worth an explanatory comment.

File: web/packages/teleport/src/stores/storeUserContext.ts [hunk 5]
```diff
@@ -191,4 +194,12 @@ export default class StoreUserContext extends Store<UserContext> {
   getAssistantAccess() {
     return this.state.acl.assist;
   }
+
+  getAccessCapabilities() {
+    return this.state.accessCapabilities;
+  }
```
Comment (ibeckermayer) [inline web/packages/teleport/src/stores/storeUserContext.ts:200]: nit: we already have

https://github.com/gravitational/teleport/blob/17d4b45c345161174f45e3b04082a7c25791dc17/web/packages/teleport/src/stores/storeUserContext.ts#L79-L81

which is all this is being used for

**Review by ibeckermayer — APPROVED** [review]

**Pull request merged**